lib/dry_twitter/sign_in/validate.rb

```diff
@@ -0,0 +1,35 @@
+require "dry/transaction/operation"
+require "dry_twitter/import"
+require 'dry-monads'
+require 'armor'
+require 'securerandom'
+
+module DryTwitter
+  module SignIn
+    class Validate
+      include Dry::Transaction::Operation
+      include DryTwitter::Import["repositories.users"]
+      include Dry::Monads::Try::Mixin
+
+      def call(input)
+        result = Try() {
```

Comment:
I agree specific exceptions are always better, but in your own custom code where / when you know what type of exceptions can be raised.

How can I know here what errors can be thrown by a custom library. I'd have to inspect their code, because I don't see this documented.

Alternatively I could drop the try block and return just a success monad from the operation.

Reply:
One way to do it is to browse the docs http://rom-rb.org/api/rom-sql/ and search for error, that should be simpler.